### PR TITLE
support specify ouput filename format

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
implement https://github.com/pingcap/dumpling/issues/114

### What is changed and how it works?
Add `--output-filename-format` argument to support specifying the format of output filename.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 
### Release note

- support specify output filename format

### CODE REVIEW

File: cmd/dumpling/main.go [hunk 1]
```diff
@@ -119,6 +120,7 @@ func main() {
 	pflag.StringVar(&keyPath, "key", "", "The path name to the client private key file for TLS connection")
 	pflag.StringVar(&csvSeparator, "csv-separator", ",", "The separator for csv files, default ','")
 	pflag.StringVar(&csvDelimiter, "csv-delimiter", "\"", "The delimiter for values in csv files, default '\"'")
+	pflag.StringVar(&outputFilenameFormat, "output-filename-format", "{{.DB}}.{{.Table}}.{{.Index}}", "The output filename format, default '{{.DB}}.{{.Table}}.{{.Index}}'")
```
Comment (kennytm) [inline cmd/dumpling/main.go:123]:
```suggestion
	pflag.StringVar(&outputFilenameFormat, "output-filename-template", "{{.DB}}.{{.Table}}.{{.Index}}", "The output filename template (without file extension), default '{{.DB}}.{{.Table}}.{{.Index}}'")
```

File: v4/export/config.go [hunk 2]
```diff
@@ -49,46 +49,48 @@ type Config struct {
...
-	EscapeBackslash   bool
-	DumpEmptyDatabase bool
-	SessionParams     map[string]interface{}
+	TableFilter          filter.Filter
+	Rows                 uint64
+	Where                string
+	FileType             string
+	EscapeBackslash      bool
+	DumpEmptyDatabase    bool
+	OutputFilenameFormat string
```
Comment (kennytm) [inline v4/export/config.go:58]: consider using the parsed `*template.Template` here so it does not need to be parsed every time we call `WriteTableData`.

File: v4/export/writer.go [hunk 3]
```diff
@@ -126,25 +139,38 @@ type csvOption struct {
...
 }
 
-func (namer *outputFileNamer) NextName() string {
-	defer func() { namer.chunkIndex++ }()
-	if namer.dbName == "" || namer.tableName == "" {
-		return fmt.Sprintf("result.%d", namer.chunkIndex)
+func (namer *outputFileNamer) NextName(tmpl *template.Template) (string, error) {
+	defer func() { namer.Index++ }()
+	if namer.DB == "" || namer.Table == "" {
+		return fmt.Sprintf("result.%d", namer.Index), nil
```
Comment (kennytm) [inline v4/export/writer.go:151]: 🤔

Comment (lichunzhu) [reply v4/export/writer.go]: How about the fix in https://github.com/pingcap/dumpling/pull/120/commits/12d0e06fbdc477d70077965249ddb6af82a66147?

File: v4/export/config.go (hunk 2 above)
Comment (lichunzhu) [reply v4/export/config.go]: addressed in https://github.com/pingcap/dumpling/pull/120/commits/50d18ad53396b887b1cbf08309eaeb153dd48638

File: cmd/dumpling/main.go (hunk 1 above)
Comment (lichunzhu) [reply cmd/dumpling/main.go]: addressed in https://github.com/pingcap/dumpling/pull/120/commits/50d18ad53396b887b1cbf08309eaeb153dd48638

File: v4/export/config.go [hunk 4]
```diff
@@ -49,48 +50,49 @@ type Config struct {
...
+	FileType           string
+	EscapeBackslash    bool
+	DumpEmptyDatabase  bool
+	OutputFileTemplate *template.Template
+	SessionParams      map[string]interface{}
 }
 
 func DefaultConfig() *Config {
 	allFilter, _ := filter.Parse([]string{"*.*"})
+	tmpl, _ := template.New("filename").Parse("{{.DB}}.{{.Table}}.{{.Index}}")
```
Comment (kennytm) [inline v4/export/config.go:65]: use `template.Must` here

**Comment by lichunzhu** [comment]
It seems that GitHub can identify my new commit. I will reopen this in another PR

**Pull request closed without merging**